python/lsst/jointcal/jointcal.py

```diff
@@ -593,34 +592,49 @@ def validate(self):
             lsst.log.warning(msg)
 
     def setDefaults(self):
-        # Use science source selector which can filter on extendedness, SNR, and whether blended
-        self.sourceSelector.name = 'science'
-        # Use only stars because aperture fluxes of galaxies are biased and depend on seeing
+        # Use only stars because aperture fluxes of galaxies are biased and depend on seeing.
         self.sourceSelector['science'].doUnresolved = True
+        self.sourceSelector['science'].unresolved.name = "extendedness"
         # with dependable signal to noise ratio.
         self.sourceSelector['science'].doSignalToNoise = True
         # Min SNR must be > 0 because jointcal cannot handle negative fluxes,
         # and S/N > 10 to use sources that are not too faint, and thus better measured.
         self.sourceSelector['science'].signalToNoise.minimum = 10.
-        # Base SNR on CalibFlux because that is the flux jointcal that fits and must be positive
-        fluxField = f"slot_{self.sourceFluxType}Flux_instFlux"
-        self.sourceSelector['science'].signalToNoise.fluxField = fluxField
-        self.sourceSelector['science'].signalToNoise.errField = fluxField + "Err"
+        # Base SNR on apFlux because that is the flux that jointcal fits and must be positive.
+        self.sourceSelector['science'].signalToNoise.fluxField = "apFlux_12_0_instFlux"
+        self.sourceSelector['science'].signalToNoise.errField = "apFlux_12_0_instFluxErr"
         # Do not trust blended sources' aperture fluxes which also depend on seeing
         self.sourceSelector['science'].doIsolated = True
+        self.sourceSelector['science'].isolated.parentName = 'parentSourceId'
+        self.sourceSelector['science'].isolated.nChildName = 'deblend_nChild'
         # Do not trust either flux or centroid measurements with flags,
         # chosen from the usual QA flags for stars)
         self.sourceSelector['science'].doFlags = True
-        badFlags = ['base_PixelFlags_flag_edge', 'base_PixelFlags_flag_saturated',
-                    'base_PixelFlags_flag_interpolatedCenter', 'base_SdssCentroid_flag',
-                    'base_PsfFlux_flag', 'base_PixelFlags_flag_suspectCenter']
+        badFlags = ['pixelFlags_edge',
+                    'pixelFlags_saturated',
+                    'pixelFlags_interpolatedCenter',
+                    'pixelFlags_interpolated',
```

Comment:
I don't know: do we have documentation about which flags are sub/supersets of which?

These are both in the list that I'm taking from DECam/Subaru, so I'd rather keep them until we determine otherwise.

Reply:
See:
https://github.com/lsst/meas_base/blob/26961ef6c7a3bd91c4b40a0307c740244fea775e/include/lsst/meas/base/PixelFlags.h#L46-L51
So the difference is the non-center ones correspond to mask planes anywhere in the footprint, whereas the `Center` ones correspond to mask bits which occur in the center of the object (so, indeed, the latter is a subset of the former).  Can you file a ticket to have someone properly assess the flags that should be in play here?

Reply:
I filed that as https://jira.lsstcorp.org/browse/DM-33298